MSMAll/MSMAllPipeline.sh

```diff
@@ -138,75 +147,174 @@ log_File_Must_Exist "${TopographyTarget}"
+    NativeMyelinMap="MyelinMap.native.dscalar.nii"    
+    # Build the average myelin map command.
+    average_cmd="${CARET7DIR}/wb_command -cifti-average \
+        $StudyFolder/$TemplateSession/MNINonLinear/Native/$TemplateSession.$NativeMyelinMap"
+        
+    for tp in "${SessionsLong[@]}"; do
+        SessionLong=$tp.long.$TemplateLong                      #longitudinal session directory name
+        average_cmd="$average_cmd -cifti $StudyFolder/$SessionLong/MNINonLinear/Native/$SessionLong.$NativeMyelinMap"
+            
+        echo "searching $SessionLong for eligible fMRI runs"
+        if [ ! -d "$StudyFolder/$SessionLong/MNINonLinear/Results" ]; then 
+            log_Err_Abort "ICAFix output does not exist for longitudinal session $SessionLong in $StudyFolder"
+        fi    
+        ResultsTPLongDir=$StudyFolder/$SessionLong/MNINonLinear/Results
+        # iterate over possible fMRI runs and build a list of found runs for this timepoint.
+        # Found runs are copied to template directory and all relevant files/dirs are renamed 
+        # for the run to be unique within subject.
+        # template-based fMRI naming pattern, assuming original run <fMRIRun> label is: <Session>_<fMRIRun>, 
+        # where <Session> is longitudinal session label. It is expected to be named <Subject>_<Visit_ID>.
+        for fmriName in "${PossibleRuns[@]}"; do
+            if [ -d "$ResultsTPLongDir/$fmriName" ]; then            
+                TemplateRun=${tp}_${fmriName}
+                echo "found $TemplateRun, copying"
+                mkdir -p $ResultsTemplateDir/$TemplateRun
+                #bulk copy
+                cp -r "$ResultsTPLongDir/$fmriName"/* "$ResultsTemplateDir/$TemplateRun/"
+                pushd "$ResultsTemplateDir/${TemplateRun}" &> /dev/null                
+                for fd in ${fmriName}_* ${fmriName}.*; do
+                    [ -e "$fd" ] || continue
+                    mv "$fd" "${fd//$fmriName/$TemplateRun}"
+                done
+                popd &> /dev/null            
+                TemplateRunsStr="${TemplateRunsStr}@$TemplateRun"
+                TimepointsStr="${TimepointsStr}@$tp"
+                fMRIRunsStr="${fMRIRunsStr}@$fmriName"
+                ConcatNamesStr="${ConcatNamesStr}@$mrfixConcatName"
+            fi
+        done
+    done
+    # Arrays with template fMRI names, matching timepoint labels, original run labels, 
+    # and per-timepoint concatenated fMRI names are stored in a configuration file under 
+    # MNINonLinear/Results/. Multiple configuration files may be used for the same 
+    # subject with different fMRI combinations or longitudinal templates.
+    conf_file="$ResultsTemplateDir/$OutConfig"
+    echo "${TemplateRunsStr/#@/}" > "$conf_file"
+    echo "${TimepointsStr/#@/}" >> "$conf_file"
+    echo "${fMRIRunsStr/#@/}" >> "$conf_file"
+    echo "${ConcatNamesStr/#@/}" >> "$conf_file"
```

Comment:
I had to look up this syntax to figure out it was "replace at beginning".  We already use the simpler "remove prefix" syntax elsewhere, so I think it is more recognizable.  If it were me, I would store them as arrays and use a subshell with `IFS` to make the string without an extra `@`...
```suggestion
    echo "${TemplateRunsStr#@}" > "$conf_file"
    echo "${TimepointsStr#@}" >> "$conf_file"
    echo "${fMRIRunsStr#@}" >> "$conf_file"
    echo "${ConcatNamesStr#@}" >> "$conf_file"
```
A demo of the `IFS` trick (note, the semicolon and `*` instead of `@` are both important - the subshell stops `IFS` from affecting other commands in the script):
```
foo=(a b c)
(IFS='+'; echo "${foo[*]}")
```